codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/StructureGenerator.kt

```diff
@@ -198,6 +198,16 @@ class StructureGenerator(
                         .write("if (other.#1L == null) return false", memberName)
                         .write("if (!#1L.contentEquals(other.#1L)) return false", memberName)
                         .closeBlock("} else if (other.#1L != null) return false", memberName)
+                } else if (target is ListShape && target.member.targetOrSelf(model).isBlobShape) {
+                    openBlock("if (#L != null) {", memberName)
+                        .write("if (other.#L == null) return false", memberName)
+                        .write("if (#1L.size != other.#1L.size) return false", memberName)
+                        .openBlock("for (i in #L.indices) {", memberName)
+                        .write("if (!#1L[i].contentEquals(other.#1L[i])) return false", memberName)
+                        .closeBlock("}")
+                        .closeBlock("} else if (other.#1L != null) return false", memberName)
+                } else if (target is DoubleShape || target is FloatShape) {
+                    write("if (#1L?.isNaN() == true && other.#1L?.isNaN() == true) { } else if (#1L != other.#1L) return false", memberName)
```

Comment:
Yes it is supposed to be empty. If both values are NaN we want to do nothing. If we `return true` here then the `equals` function will exit early. We can only `return false` when something is invalid.

Example:
```kt
override fun equals(other: #Q?): #Q { 
	if (this === other) return true	
	if (other == null || this::class != other::class) return false
	other as #T

	if (myBoolean != other.myBoolean) return false
    if (myInt != other.myInt) return false
    if (myFloat?.isNaN() == true && other.myFloat?.isNaN() == true) { } else if (myFloat != other.myFloat) return false
	if (myByte != other.myByte) return false
}
```

If we had a `return true` in that empty block, the function would exit early and the `myByte` comparison would not happen.

I was not really happy with the way this conditional looks, but couldn't think of a better way to format it. Do you have any ideas?

Reply:
I think we can just get rid of that if statement since it does nothing. If this statement: `if (myFloat != other.myFloat)` requires the values to not be `NaN` then it can be included in the statment

Reply:
It needs to be there because "`Float.NaN == Float.NaN` always returns `false`"

Reply:
Can it be changed to something like?
```kotlin
val bothAreNan = myFloat?.isNaN() == true && other.myFloat?.isNaN() == true
if ( !bothAreNan && myFloat != other.myFloat)
```
I think it accomplishes the same thing. I feel like a comment in the code would be useful as well in this case.

Reply:
I learned we can use `.equals()` to compare NaNs, so I simplified it to just do that.